components/NewMeetingForm/NewMeetingForm.test.js

```diff
@@ -1,21 +1,86 @@
 import NewMeetingForm from "./NewMeetingForm";
 import React from "react";
 import { render, screen, fireEvent } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
 import { useRouter } from "next/router";
 
+const defaultData = {
+  id: "0",
+  location: "Volkspark",
+  date: "2023-04-14",
+  time: "15:00",
+  furtherInfo: "Nur bei gutem Wetter!",
+  attending: "Fiete",
+};
+
 jest.mock("next/router", () => ({
   useRouter: jest.fn(),
 }));
 
-test("checks if submit handler is called on form submit", async () => {
+test("submit handler is called on form submit", async () => {
   useRouter.mockImplementation(() => ({
     push: jest.fn(),
   }));
   const onSubmit = jest.fn();
 
   render(<NewMeetingForm onSubmit={onSubmit} />);
 
-  const form = screen.getByRole("form");
+  const form = screen.getByTestId("form");
   fireEvent.submit(form);
   expect(onSubmit).toHaveBeenCalled();
 });
+
+test("submits edited form data when fields are altered", async () => {
+  const onSubmit = jest.fn();
+  render(
+    <NewMeetingForm
+      onSubmit={onSubmit}
+      formTitle="Angaben ändern?"
+      defaultData={defaultData}
+    />
+  );
+
+  const user = userEvent.setup();
+
+  const location = screen.getByTestId("location");
+  const date = screen.getByTestId("date");
+  const time = screen.getByTestId("time");
+  const attending = screen.getByTestId("attending");
+  const urueden = screen.getByTestId("unkastrierterueden");
+  const krueden = screen.getByTestId("kastrierterueden");
+  const uhuendinnen = screen.getByTestId("unkastriertehuendinnen");
+  const khuendinnen = screen.getByTestId("kastriertehuendinnen");
+  const lhuendinnen = screen.getByTestId("laeufigehuendinnen");
+  const welpen = screen.getByTestId("welpen");
+  const form = screen.getByTestId("form");
+
+  await userEvent.clear(location);
+  await user.type(location, "Elbstrand");
+  await userEvent.clear(attending);
+  await user.type(attending, "Fiete, Lore");
+  fireEvent.change(date, "2020-05-24");
+  fireEvent.change(time, "16:00");
+  fireEvent.click(urueden);
+  fireEvent.click(krueden);
+  fireEvent.click(uhuendinnen);
+  fireEvent.click(khuendinnen);
+  fireEvent.click(lhuendinnen);
+  fireEvent.click(welpen);
+
+  fireEvent.submit(form);
```

Comment:
I am not sure if this is true, but I think submit can't be a userEvent.. in such a case it should rather be a click event on the submit button. With this and some other changes to the syntax the test finally passed. Thanks for taking such a thorough look at my code!

Reply:
I tried your code.  
1. Because there are already default values in "date" and "time", the `.type` works after using `.clear`.
2. Clicking the submit button will submit the form. I saw this way from a demo .  
In my notebook the test passes with those code.
```javascript
 await userEvent.clear(location);
  await user.type(location, "Elbstrand");
  await userEvent.clear(attending);
  await user.type(attending, "Fiete, Lore");
  await user.clear(date);
  await user.type(date, "2023-05-24");
  await user.clear(time);
  await user.type(time, "16:00");
  await user.click(urueden);
  await user.click(krueden);
  await user.click(uhuendinnen);
  await user.click(khuendinnen);
  await user.click(lhuendinnen);
  await user.click(welpen);

  const button = screen.getByRole("button");
  await user.click(button);
```